pkg/handler/metricsrecorder/handler.go

```diff
@@ -0,0 +1,338 @@
+package metricsrecorder
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/observeinc/aws-sam-apps/pkg/logging"
+
+	"net/http"
+
+	"github.com/go-logr/logr"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+)
+
+type GraphQLRequest struct {
+	Query string `json:"query"`
+}
+
+type Handler struct {
+	Logger           logr.Logger
+	MetricStreamName string
+	FirehoseArn      string
+	RoleArn          string
+	OutputFormat     string
+	AccountNumber    string
+	DatasourceID     string
+	SecretName       string
+}
+
+type Config struct {
+	MetricStreamName string `env:"METRIC_STREAM_NAME,required"`
+	AccountNumber    string `env:"ACCOUNT_NUMBER,required"`
+	DatasourceID     string `env:"DATASOURCE_ID,required"`
+	SecretName       string `env:"SECRET_NAME,required"`
+	FirehoseArn      string `env:"FIREHOSE_ARN,required"`
+	RoleArn          string `env:"ROLE_ARN,required"`
+	OutputFormat     string `env:"OUTPUT_FORMAT,required"`
+	Logging          *logging.Config
+}
+
+type MetricsListItem struct {
+	Namespace   string   `json:"namespace"`
+	MetricNames []string `json:"metricNames"`
+}
+
+func convertToMetricStreamFilters(MetricsList []MetricsListItem) []types.MetricStreamFilter {
+
+	if len(MetricsList) == 0 {
```

Comment:
dont think we need to handle the 0 care do we? just initalize the slice as empty and then the range will just not run if its empty

Reply:
Agreed, there was some weird error with the types being correct, and this was my strong-armed way of fixing it, but your way is more clear.